app/javascript/vue/repository_item_sidebar/Textarea.vue

```diff
@@ -0,0 +1,151 @@
+    };
+  },
+  props: {
+    expandable: { type: Boolean, required: true },
+    collapsed: { type: Boolean, required: true },
+    initialValue: String,
+    noContentPlaceholder: String,
+    placeholder: String,
+    decimals: { type: Number, default: null },
+    unEditableRef: { type: String, required: true },
+    smartAnnotation: { type: Boolean, default: false },
+    sa_value: { type: String },
+  },
+  mounted() {
+    this.value = this.initialValue;
+    this.$nextTick(() => {
+      this.toggleExpandableState();
+    });
+  },
+  beforeUpdate() {
+    if (!this.$refs.textareaRef) return;
+
+    if (this.decimals !== null) this.validateNumberInput();
+  },
+  watch: {
+    initialValue: {
+      handler() {
+        this.value = this.initialValue;
+        this.toggleExpandableState();
+      },
+      deep: true,
+    },
+    editing() {
+      this.$nextTick(() => {
+        if (this.editing) {
+          this.setCaretAtEnd();
+          this.refreshTextareaHeight();
+          return;
+        }
+
+        this.toggleExpandableState();
+      })
+    },
+  },
+  computed: {
+    canEdit() {
+      return this.permissions?.can_manage && !this.inArchivedRepositoryRow;
+    }
+  },
+  methods: {
+    handleKeydown(event) {
+      if (event.key === 'Enter') {
+        event.preventDefault();
+        this.$refs.textareaRef.blur();
+      }
+    },
+    handleBlur() {
+      if ($('.atwho-view:visible').length) return;
+
+      if (this.smartAnnotation) {
+        this.value = this.$refs.textareaRef.value.trim() // Fix for smart annotation
+      }
+      this.editing = false;
+      this.toggleExpandableState();
+      this.$emit('update', this.value);
+    },
+    toggleExpandableState() {
+      this.$nextTick(() => {
+        if (!this.$refs[this.unEditableRef]) return;
+
+        const maxCollapsedHeight = '80';
+        const scrollHeight = this.$refs[this.unEditableRef].scrollHeight;
+        const expandable = scrollHeight > maxCollapsedHeight;
+        this.$emit('toggleExpandableState', expandable);
+      });
+    },
+    enableEdit(e) {
+      if (e && $(e.target).hasClass('atwho-user-popover')) return;
+      if (e && $(e.target).hasClass('sa-link')) return;
+      if (e && $(e.target).parent().hasClass('atwho-inserted')) return;
+
+      this.editing = true;
+      this.$nextTick(() => {
+        if (this.smartAnnotation) {
+          SmartAnnotation.init($(this.$refs.textareaRef), false);
+        }
+      });
+    },
+    refreshTextareaHeight() {
+      this.$nextTick(() => {
+        if (!this.editing) return;
+        const textarea = this.$refs.textareaRef;
+        textarea.style.height = '0px';
+        // 16px is the height of the textarea's line
+        textarea.style.height = textarea.scrollHeight - 16 + 'px';
+      });
+    },
+    setCaretAtEnd() {
+      this.$nextTick(() => {
+        if (!this.editing) return;
+
+        this.$refs.textareaRef.focus();
+      });
+    },
+    validateNumberInput() {
```

Comment:
should probably be called something more like "enforceNumberIntput", as validate implies it sets/displays some errors